### DISCUSSION TITLE
Open Source Contribution

### DISCUSSION THREAD

**Discussion by shashankbezgam** [seerge/g-helper#3135] [Q&A]
I would like to know if there are any specific requirements for contributing to this project. I do understand that overloading the app is something you aren't looking for. I have an idea that is extremely light addition to the existing app. I would like to make this contribution.

**Comment by seerge** [comment]
@shashankdatta hello, 

I usually try to extend the app with a feature if it is : 

- Proprietary hardware or firmware setting and you can't set it anyhow else except using asus apps (GPU modes, Performanc Modes, Fan settings, etc)
- Software thing that Armoury Crate already does (Visual Modes, Flicker-free dimming, etc)  
- It's something that will make life easier for lots of people (FN-Lock, CPU boost control) 

But I prefer to avoid additions that affect only one user from a thousand, but will still take place in the UI as a checkbox or so :) 

What exactly do you want to add ?

**Reply by shashankbezgam** [reply]
My idea is simple and would definitely help everyone like you mentioned.

I am not sure if it's already in works but I wanted to add a simple checkbox/button on the ui for preventing the laptop from sleeping. 

Basically a Caffeinate option. I don't want to install other apps that run in the background just for occasional use. But it's definitely something I would love to have as a developer/gamer/daily user.

**Reply by seerge** [reply]
@shashankdatta hmm, but what is the practical purpose of this ? Isn't it easier just to disable sleep in windows settings directly ? :)

**Comment by shashankbezgam** [comment]
There are many situations where one might want the laptop to stay on
indefinitely (until a certain time or until interrupted), and it becomes a
hassle to always modify the sleep settings just to turn off sleep
temporarily.

 For instance, when installing games, running programs like ML model
training, leaving a movie on pause to do some work, or taking an online
exam where we might shift focus for a few minutes, sleep mode would disrupt
progress. Other scenarios include downloading large files, remote desktop
sessions, rendering videos, hosting a server, and monitoring systems.

**Reply by seerge** [reply]
Hmm, as for me this is still quite exotic thing. + there are multiple existing solutions for that like [this](https://apps.microsoft.com/detail/9nblggh4whsp?hl=en-us&gl=US) or [this](https://www.softwareok.com/?seite=Microsoft/DontSleep) or [this](https://www.zhornsoftware.co.uk/caffeine/) :)

**Comment by shashankbezgam** [comment]
Yes, these tools do exist, but installing them separately for such a small
feature (with a lot of uses) isn’t worth having them run in the background.
The only reason I thought of G-helper is that it always needs to be running
in the background, and I don’t mind it using resources. Simple already
existing features like fn-lock and one-time 100% charge are what motivated
me to add this.

However, I do understand your reasoning too. But I will still develop this
on my fork and have it ready, and maybe you can take a look at it later.

**Comment by shashankbezgam** [comment]
@seerge I've implemented the Caffeinated feature we discussed a year ago in PR #3945. 

The implementation:
- Prevents system sleep/display timeout for a configurable duration
- Uses only native Windows APIs (SetThreadExecutionState)
- Integrates with G-Helper's existing keyboard shortcuts system
- Follows G-Helper's UI/UX design patterns

I understand there are standalone tools that offer similar functionality, but with G-Helper's growing userbase across various ASUS products (not just gaming laptops), incorporating this feature directly helps users avoid installing additional background apps for such a simple but useful function.

I'd appreciate if you could take a look at the PR and provide any feedback. The feature is ready to use and includes full localization support.

Link to PR: https://github.com/seerge/g-helper/pull/3945

**Reply by visallur** [reply]
Hey great feature man, I was looking forward to this! Serves the best for my usecase and for a lot of other people as well in my opinion.

**Reply by shashankbezgam** [reply]
> Hey great feature man, I was looking forward to this! Serves the best for my usecase and for a lot of other people as well in my opinion.

Exactly! It's such a simple yet handy feature, I hope the devs reply to my implementation with their suggestions so I can take this forward

**Reply by Nikhil-Koganti** [reply]
Hey @shashankdatta! This feature looks easy and intuitive to use. Definitely useful for me as a college student who runs ML models in the background. I also install large storage games that take a long while to install. I always have to fiddle with power saving settings for this. But, this makes it much easier. Appreciate it man!

**Reply by shashankbezgam** [reply]
> Hey @shashankdatta! This feature looks easy and intuitive to use. Definitely useful for me as a college student who runs ML models in the background. I also install large storage games that take a long while to install. I always have to fiddle with power saving settings for this. But, this makes it much easier. Appreciate it man!

Yes! Nw. Ty for the support. Hopefully it goes somewhere

**Reply by MohammadTawfeeqjeelani** [reply]
Hey @shashankdatta, this feature looks really helpful for day-to-day use especially for someone like me, a student who often uses their laptop without a power supply. My laptop tends to go to sleep frequently, and this usually happens when I’m watching videos or read lecture notes that are paused for a while, which ends up disrupting my study flow. This feature in g helper could definitely help me avoid that issue.

**Reply by visallur** [reply]
@seerge hope you look into this a bit more and consider implementing it :)
